.github/workflows/ci_tests.yaml

```diff
@@ -117,6 +117,14 @@ jobs:
         shell: bash -l {0}
         run: make test PYTEST_EXTRA="-r P"
 
+      # Upload diff images on test failure
+      - name: Upload diff images if any test fails
+        uses: actions/upload-artifact@v2
+        if: ${{ failure() }}
+        with:
+          name: artifact-${{ matrix.os }}-${{ matrix.python-version }}
```

Comment:
It works as expected now. 

![image](https://user-images.githubusercontent.com/3974108/97264604-a658ce00-17fb-11eb-962d-a4467b9c9234.png)

What about changing the names to:
```suggestion
          name: artifact-${{ runner.os }}-Python${{ matrix.python-version }}
```

[`runner.os`](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#runner-context)'s possible values are Linux, Windows, or macOS. 

Then the artifacts will be `artifact-Windows-Python3.8`.

Reply:
> [`runner.os`](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#runner-context)'s possible values are Linux, Windows, or macOS.
> 
> Then the artifacts will be `artifact-Windows-Python3.8`.

Oh yes, shorter name is good! I'll also add this to the GMT Latest tests yaml file later.

Reply:
Alright, works on GMT Latest/master tests as well:

![image](https://user-images.githubusercontent.com/23487320/97363111-efe20f80-1906-11eb-90b0-981d48a0228b.png)